src/main/java/com/xebialabs/overthere/CmdLineArgument.java

```diff
@@ -117,34 +110,25 @@ protected void encodeString(String str, OperatingSystemFamily os, StringBuilder
 
         switch (os) {
             case WINDOWS:
-                if (!containsAny(str, SPECIAL_CHARS_WINDOWS)) {
+                if (!WindowsCommandLineArgsSanitizer.containsAnySpecialChars(str)) {
                     builder.append(str);
                 } else {
                     encodeArgumentWithSpecialCharactersForWindows(str, builder);
                 }
                 break;
             case UNIX:
             case ZOS:
-                if (!containsAny(str, SPECIAL_CHARS_UNIX)) {
+                if (!UnixCommandLineArgsSanitizer.containsAnySpecialChars(str)) {
                     builder.append(str);
                 } else {
-                    encodeArgumentWithSpecialCharactersForUnix(str, builder);
+                    builder.append(UnixCommandLineArgsSanitizer.sanitize(str));
                 }
                 break;
             default:
                 throw new RuntimeException("Unknown os " + os);
         }
     }
 
-    private static boolean containsAny(String str, String chars) {
-        for (char c : chars.toCharArray()) {
-            if (str.indexOf(c) >= 0) {
-                return true;
-            }
-        }
-        return false;
-    }
-
     private static void encodeArgumentWithSpecialCharactersForWindows(String str, StringBuilder builder) {
```

Comment:
Windows is already escaping with quotes and that doesn't work specifically for winrs , unix code used the logic that has been moved to the UnixSantizer